genesis/process/sovereignGenesisBlockCreator.go

```diff
@@ -65,3 +76,204 @@ func (gbc *sovereignGenesisBlockCreator) createSovereignEmptyGenesisBlocks() (ma
 
 	return mapEmptyGenesisBlocks, nil
 }
+
+func (gbc *sovereignGenesisBlockCreator) computeSovereignDNSAddresses(enableEpochsConfig config.EnableEpochs) error {
+	initialAddresses, err := factory.DecodeAddresses(gbc.arg.Core.AddressPubKeyConverter(), gbc.arg.DNSV2Addresses)
+	if err != nil {
+		return err
+	}
+
+	return gbc.computeDNSAddresses(enableEpochsConfig, initialAddresses)
+}
+
+func (gbc *sovereignGenesisBlockCreator) createSovereignHeaders(args *headerCreatorArgs) (map[uint32]data.HeaderHandler, error) {
+	shardID := core.SovereignChainShardId
+	log.Debug("sovereignGenesisBlockCreator.createHeaders", "shard", shardID)
+
+	var genesisBlock data.HeaderHandler
+	var scResults [][]byte
+	var err error
+
+	genesisBlock, scResults, gbc.initialIndexingData[shardID], err = createSovereignShardGenesisBlock(
+		args.mapArgsGenesisBlockCreator[shardID],
+		args.mapBodies[shardID],
+		args.nodesListSplitter,
+		args.mapHardForkBlockProcessor[shardID],
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("'%w' while generating genesis block for shard %d", err, shardID)
+	}
+
+	genesisBlocks := make(map[uint32]data.HeaderHandler)
+	allScAddresses := make([][]byte, 0)
+	allScAddresses = append(allScAddresses, scResults...)
+	genesisBlocks[shardID] = genesisBlock
+	err = gbc.saveGenesisBlock(genesisBlock)
+	if err != nil {
+		return nil, fmt.Errorf("'%w' while saving genesis block for shard %d", err, shardID)
+	}
+
+	err = gbc.checkDelegationsAgainstDeployedSC(allScAddresses, gbc.arg)
+	if err != nil {
+		return nil, err
+	}
+
+	gb := genesisBlocks[shardID]
+	log.Info("sovereignGenesisBlockCreator.createHeaders",
```

Comment:
sovereignGenesisBlockCreator.createSovereignHeaders

Reply:
Fixed it